pkg/securitycontext/provider.go

```diff
@@ -65,7 +65,9 @@ func (p SimpleSecurityContextProvider) ModifyHostConfig(pod *v1.Pod, container *
 			for _, group := range pod.Spec.SecurityContext.SupplementalGroups {
 				hostConfig.GroupAdd = append(hostConfig.GroupAdd, strconv.Itoa(int(group)))
 			}
-			if pod.Spec.SecurityContext.FSGroup != nil {
+			// only add the FSGroup to the config if we are not in a remap environment.  For remap environments
+			// is is assumed that the root user is remapped to the FSGroup on the host.
+			if pod.Spec.SecurityContext.FSGroup != nil && !isRemap {
```

Comment:
I think for containers to run as they do without userns, the group should be added for userns as well. However, based on default docker remap, only 65536 IDs are mapped so group would have to be in that range. If a higher range is desired then docker should be configured accordingly.